build_library/build_image_util.sh

```diff
@@ -345,12 +394,41 @@ write_licenses() {
             echo -n "\"${lic}\""
         done
 
-        echo -n "]}"
+        echo -n "], \"description\": \"${description}\", \"homepage\": \"${homepage}\", \"source\": \"${src_uri}\", \"files\": \"${files}\"}"
     done >> "$2"
 
     echo -e "\n]" >> "$2"
 }
 
+# Convert the license JSON to a text format and store it and all licenses in the rootfs.
+# The generation of the metadata in write_licenses takes a while because the invocation of the portage wrapper
+# is slow etc., thus, just reuse the JSON data which is also convenient with "jq" string substitution.
+insert_licenses() {
+    local json_input="$1"
+    local root_fs_dir="$2"
+    local output="${root_fs_dir}"/usr/share/licenses/INFO
+    sudo mkdir -p "${root_fs_dir}"/usr/share/licenses
+    echo "Flatcar Container Linux distributes software from various projects." | sudo_clobber "${output}"
+    echo "The following list of projects shows their licenses, how to obtain the source code," | sudo_append "${output}"
+    echo "and which binary files in Flatcar Container Linux were created from it." | sudo_append "${output}"
+    echo "The license texts are available under /usr/share/licenses/common/ (read with \"less NAME.gz\")." | sudo_append "${output}"
+    echo "Build system files and patches used to build these projects are located at:" | sudo_append "${output}"
+    echo "https://github.com/flatcar-linux/coreos-overlay/" | sudo_append "${output}"
+    echo "https://github.com/flatcar-linux/portage-stable/" | sudo_append "${output}"
+    echo "https://github.com/flatcar-linux/scripts/" | sudo_append "${output}"
+    echo "Information on how to build Flatcar Container Linux can be found under:" | sudo_append "${output}"
+    echo "https://docs.flatcar-linux.org/os/sdk-modifying-flatcar/" | sudo_append "${output}"
+    echo | sudo_append "${output}"
```

Comment:
Why are you calling sudo_append after each echo? It makes this very hard to read... Any reason why you're not using a heredoc?

Reply:
Yeah, heredoc would also work.

Reply:
Done